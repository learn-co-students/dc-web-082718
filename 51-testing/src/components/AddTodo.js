import React from "react";
import { connect } from "react-redux";
import { addTodo } from "../redux/actions";

// controlled component version - both work!

class AddTodo extends React.Component {
  state = {
    inputText: ""
  };

  handleInputChange = e => {
    this.setState({
      inputText: e.target.value
    });
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.addTodo(this.state.inputText);
    this.setState({
      inputText: ""
    });
  };

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <input
            placeholder="Add a New Todo..."
            className="ui input"
            onChange={this.handleInputChange}
            value={this.state.inputText}
          />
        </form>
      </div>
    );
  }
}

// OR... //

// functional compoent -- both work!
// const AddTodo = ({ addTodo }) => {
//   // the 'addTodo' prop is provided by the 'connect' wrapper at the end of the file
//   let input;
//   return (
//     <div>
//       <form
//         onSubmit={event => {
//           event.preventDefault();
//           addTodo(input.value);
//           input.value = "";
//         }}
//       >
//         <input
//           ref={node => {
//             input = node;
//           }}
//         />
//         <button type="submit"> Add Todo </button>
//       </form>
//     </div>
//   );
// };

// connect and map addTodo to props
// when we pass this function (as the second argument) to connect,
// it provides the AddTodo component with a *prop* that
// dispatches the addTodo action to the reducer, along with the value of the user's input - the 'text' of the todo

const mapDispatchToProps = dispatch => {
  return {
    addTodo: inputText => dispatch(addTodo(inputText))
  };
};

// remember that mapDispatchToProps is the 2nd argument to connect -
// if this component needed some other data from the store, then the first space
// (where null is currently passed) would handle this
export default connect(
  null,
  mapDispatchToProps
)(AddTodo);

export { AddTodo };
