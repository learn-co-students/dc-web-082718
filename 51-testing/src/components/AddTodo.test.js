import React from "react";
import { shallow } from "enzyme";
import { AddTodo } from "./AddTodo";

describe("add todo component", () => {
  it("matches its snapshot", () => {
    const wrapper = shallow(<AddTodo />);
    expect(wrapper).toMatchSnapshot();
  });
});
