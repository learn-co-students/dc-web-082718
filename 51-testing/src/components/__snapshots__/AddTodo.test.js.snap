// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add todo component matches its snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="ui input"
      onChange={[Function]}
      placeholder="Add a New Todo..."
      value=""
    />
  </form>
</div>
`;
